Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,163 @@
+	[ -d "/etc/opt/microsoft/dependency-agent" ] && LogErr "Directory /etc/opt/microsoft/dependency-agent exists." && exit 1
+	[ -d "/lib/microsoft-dependency-agent" ] && LogErr "Directory /lib/microsoft-dependency-agent exists." && exit 1
+	[ -d "/etc/init.d/microsoft-dependency-agent" ] && LogErr "Directory /etc/init.d/microsoft-dependency-agent exists." && exit 1
+
+	LogMsg "Preconditions met"
+}
+
+function download_da_linux_installer() {
+	LogMsg "Download Dependency Agent Linux installer"
+
+	curl -L https://aka.ms/dependencyagentlinux -o InstallDependencyAgent-Linux64.bin
+
+	chmod +x InstallDependencyAgent-Linux64.bin
+}
+
+function verify_install_da() {
+	./InstallDependencyAgent-Linux64.bin -vme
+
+	ret=$?
+	if [ $ret -ne 0 ]; then
+	        LogErr "Install failed with exit code ${ret}"
+	        exit 1
+	else
+	        LogMsg "Installed with exit code 0"
+	fi
+
+	[ ! -f "/opt/microsoft/dependency-agent/uninstall" ] && LogErr "/opt/microsoft/dependency-agent/uninstall does not exist." && exit 1
+	[ ! -f "/etc/init.d/microsoft-dependency-agent" ] && LogErr "/etc/init.d/microsoft-dependency-agent does not exist." && exit 1
+	[ ! -f "/var/opt/microsoft/dependency-agent/log/install.log" ] && LogErr "/var/opt/microsoft/dependency-agent/log/install.log does not exist." && exit 1
+
+	bin_version=$(./InstallDependencyAgent-Linux64.bin --version | awk '{print $5}')
+	install_log_version=$(awk "NR==3" /var/opt/microsoft/dependency-agent/log/install.log | grep 'Dependency Agent version.revision'|cut -f2 -d ":")
+	if [ "$bin_version" -ne "$install_log_version" ]; then
+	        LogErr "Version mismatch between bin version and install log version"
+	        exit 1
+	else
+	        LogMsg "Version matches between bin version and install log version"
+	fi
+
+	LogMsg "Install tests passed successfully"
+}
+
+function verify_uninstall_da() {
+
+	bash /opt/microsoft/dependency-agent/uninstall
+	
+	ret=$?
+	if [ $ret -ne 0 ]; then
+	        LogErr "Uninstall failed with exit code ${ret}"
+	        exit 1
+	else
+	        LogMsg "Uninstalled with exit code 0"
+	fi
+
+	[ -d "/opt/microsoft/dependency-agent" ] && LogErr "Directory /opt/microsoft/dependency-agent exists." && exit 1
+	[ -d "/var/opt/microsoft/dependency-agent" ] && LogErr "Directory /var/opt/microsoft/dependency-agent exists." && exit 1
+	[ -d "/etc/opt/microsoft/dependency-agent" ] && LogErr "Directory /etc/opt/microsoft/dependency-agent exists." && exit 1
+	[ -d "/lib/microsoft-dependency-agent" ] && LogErr "Directory /lib/microsoft-dependency-agent exists." && exit 1
+	[ -d "/etc/init.d/microsoft-dependency-agent" ] && LogErr "Directory /etc/init.d/microsoft-dependency-agent exists." && exit 1
+
+	LogMsg "Uninstall tests passed successfully"
+}
+
+function setup_network_trace() {
+    LogMsg "Setup network trace"
+	watch -n 5 curl https://microsoft.com &>/dev/null &
+	watch_pid=$!
```

Comment:
Once this is started in background, it will continue running until killed, even if the bash shell running the test exits or is killed.

Recommendation: Use the `trap` statement early in the script to setup a clean up "hook" to do things like killing this process if it is still running.

Reply:
Done. Added trap to call clean up function when exited or interrupted